src/Common/ZooKeeper/ZooKeeperImpl.cpp

```diff
@@ -446,8 +446,16 @@ void ZooKeeper::connect(
                 }
                 connected = true;
 
-                if (connected_callback.has_value())
-                    (*connected_callback)(i, node);
```

Comment:
Let's keep the callback instead of parsing the connection string once again

Reply:
First, the index value `i` is wrong since the `nodes` have already been shuffled.
Second, I think it's better to return the original value of host instead of ip address, and it's the easy way.
Third, my refactoring optimized the code, especially by reducing unwanted class member variables.

What do you think?